griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,160 @@
+            append_index = content[-1]['endIndex']
+            if last_text.endswith("\n"):
+                append_index -= 1
+
+            print("Appending at index:", append_index)
+
+            requests = [
+                {'insertText': {
+                    'location': {'index': append_index},
+                    'text': text
+                }}
+            ]
+
+            doc = service.documents().batchUpdate(documentId=values["document_id"],
+                                                  body={'requests': requests}).execute()
+            return InfoArtifact("Text appended successfully")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error appending text to Google Doc: {e}")
+
+    def prepend_text(self, params: dict, text: str) -> InfoArtifact | ErrorArtifact:
+        values = params["values"]
+        try:
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            doc = service.documents().get(documentId=values["document_id"]).execute()
+            if len(doc['body']['content']) == 1:
+                requests = [
+                    {'insertText': {
+                        'location': {'index': 1},
+                        'text': text
+                    }}
+                ]
+            else:
+                start_index = doc['body']['content'][1]['startIndex']
+                requests = [
+                    {'insertText': {
+                        'location': {'index': start_index},
+                        'text': text
+                    }}
+                ]
+
+            doc = service.documents().batchUpdate(documentId=values["document_id"],
+                                                  body={'requests': requests}).execute()
+            return InfoArtifact("Text prepended successfully")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error prepending text to Google Doc: {e}")
+
+    def upload_document(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        from googleapiclient.http import MediaFileUpload
+        values = params['values']
+        try:
+            service = self._build_client(self.DRIVE_SCOPES, "v3", "drive")
+            file_metadata = {
+                'name': values['file_name'],
+                'mimeType': 'application/vnd.google-apps.document'
+            }
+            media = MediaFileUpload(values['file_path'], mimetype=values['mime_type'])
+            file = service.files().create(body=file_metadata, media_body=media, fields='id').execute()
+            return InfoArtifact(f'A document was successfully uploaded with ID {file.get("id")}.')
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error uploading document to Google Docs: {e}")
+
+    def get_file_id_from_path(self, path: str) -> str | None:
```

Comment:
Make protected for consistency with Google Sheets [_get_file_id_from_path](https://github.com/griptape-ai/griptape/pull/252/files#diff-512a6c9927490e1a440658e5126ff850b51459708dbcbdc66cee0f9c4aecce39R104)

Reply:
Done!